staging/src/k8s.io/apiserver/pkg/endpoints/filters/storageversion.go

```diff
@@ -69,6 +73,31 @@ func WithStorageVersionPrecondition(handler http.Handler, svm storageversion.Man
 			handler.ServeHTTP(w, req)
 			return
 		}
+		// The system namespace is required for apiserver-identity lease to exist. Allow the apiserver
+		// itself to create namespaces.
+		// NOTE: with this exception, if the bootstrap client writes namespaces with a new version,
+		// and the upgraded apiserver dies before updating the StorageVersion for namespaces, the
+		// storage migrator won't be able to tell these namespaces are stored in a different version in etcd.
+		// Because the bootstrap client only creates system namespace and doesn't update them. This can
+		// only happen if the upgraded apiserver is the first apiserver that kicks off namespace creation,
+		// or if a upgraded server that joins an existing cluster have new system namespaces (other
+		// than kube-system, kube-public, kube-node-lease) that need to be created.
+		u, hasUser := request.UserFrom(ctx)
+		if requestInfo.APIGroup == "" && requestInfo.Resource == "namespaces" &&
+			requestInfo.Verb == "create" && hasUser &&
+			u.GetName() == user.APIServerUser && contains(u.GetGroups(), user.SystemPrivilegedGroup) {
```

Comment:
Just checking the name is sufficient, I think.

I think before beta we need a way for a priviliged user to override this check, perhaps via query parameter or header, to recover a stuck cluster, should there be some dependency chain we didn't know about.

Reply:
will add to beta requirement